ood-portal-generator/lib/ood_portal_generator/application.rb

```diff
@@ -21,12 +21,25 @@ def template
         Pathname.new(@template || OodPortalGenerator.root.join("templates", "ood-portal.conf.erb"))
       end
 
+      # The erb template file that will be rendered
+      # @return [Pathname] path to erb template
+      def maintenance_template
+        Pathname.new(@maintenance_template || OodPortalGenerator.root.join("templates", "maintenance.html.erb"))
+      end
+
       # The io object that the rendered template will be written to
       # @return [Pathname, IO] io object where rendered output is written to
       def output
         @output ? Pathname.new(@output) : $stdout
       end
 
+      # The io object that the rendered maintenance HTML wil be written to
+      # @return [Pathname, IO] io object where rendered HTML output is written to
+      def maintenance_output
+        return Pathname.new(File.join(context['public_root'], 'maintenance.html')) if context['public_root']
+        Pathname.new('/var/www/ood/public/maintenance.html')
```

Comment:
in this case its probably more appropriate to just use `if else end` instead of `return if default`

Reply:
The return statement isn't the default, it only gets hit if someone puts `public_root` into `ood_portal.yml`. Still want `if else end` ?

Reply:
Yeah its fine to have the default be in the else and the edge case be in the if.

Reply:
This change was made

Reply:
Oh there is a simpler way

```ruby
def maintenance_output
  Pathname.new(context.fetch('public_root', '/var/www/ood/public/')).join('maintenance.html')
end
```

or

```ruby
def maintenance_output
  Pathname.new(context['public_root'] || '/var/www/ood/public/').join('maintenance.html')
end
```